doc/trajopt_planner/trajopt_planner_tutorial.rst

```diff
@@ -0,0 +1,119 @@
+
+  - *STATIONARY*: the initialization matrix has joint values of the current state for all timestpes.
+
+  - *JOINT_INTERPOLATED*: the initialization matrix is a trajectory interpolated between the current state and the joint state that the user provides for ``data`` member.
+
+  - *GIVEN_TRAJ*: the user provides the entire trajectory for ``data`` member.
+
+- **TermInfo**: This is the base struct for all types of COST and CNT functions that are carried by ``cost_infos`` and ``cnt_infos`` members. COST functions are the objectives that are supposed to be minimized and CNT are the ones that must be satisfied. The current implementation containts ``JointPoseTermInfo``, ``JointVelTermInfo`` (hard-coded) and ``CartPoseTermInfo`` (is partially implemented). Member *term_type* dictates the type of the term we are adding; it could be ``TT_COST`` or ``TT_CNT`` which means a cost term or constraint term respectively. Also ``TT_USE_TIME`` can be selected for this member which allows time parameterization. In this case *use_time* of ``BasicInfo`` should be set to ``true`` as well.  The other parameters of these terms which need to be set are loaded from ``trajopt_planning.yaml`` file. The following list describes these parametrs:
+
+  - *coeffs*: weight coefficients for joints
+
+  - *targets*: the values of the joints at the constraint
+
+  - *upper_tols*: the upper limits for joint values at the constraint.
+
+  - *lower_tols*: the lower limits for joint values at the constraint.
+
+  - *first_step*: the first step that is the term is applied to.
+
+  - *last_step*: the last step that is the term is applied to.
+
+Moreover, the algorithm needs parameters specific to ``BasicTrustRegionSQP`` which are defined in a yaml file under *trajopt_param*.
+
+The following flowchart illustrates how the problem gets constructed:
+
+.. image:: trajopt.png
+   :width: 700px
+
+
+How to use TrajOpt
+------------------
+In this section, we describe how to convert the request in MoveIt to TrajOpt constraints. The first important point is that in TrajOpt, we can have the start state to be different than current state. Basically, any joint state that we want the robot pass through is just a constraint that should be added to the objective function. This joint state can be received from start state in MoveIt ``MotionPlanRequest`` or the current state from Planning Scene. The designed architecture for TrajOpt in MoveIt creates *JointPoseTermInfo* for all the constraints in *goal_constraints* and *start_state* of ``MotionPlanRequest``. On the other hand, current state of the robot also can be a constraint to add as a *JointPoseTermInfo* if the user wants the robot to start from the current state. Look at the following figure for a an example with two goal constraints:
```

Comment:
Is this actually different from the rest of the MoveIt planners? It is better to focus on just documenting TrajOpt

Reply:
I think it is? At least the `start_fixed` param is pretty different, most planners assume there's a single valid start state.

Reply:
I think the fact that current state can be different from start state should be emphasized as it might be confusing for the users.